src/main/java/com/educare/unitylend/controller/BorrowRequestController.java

```diff
@@ -57,58 +95,73 @@ ResponseEntity<List<BorrowRequest>> getBorrowRequestForUserId(@PathVariable Stri
      * @throws ControllerException If an error occurs during the borrow request retrieval process.
      */
-    @GetMapping("/get-borrow-request-in-community-by-amount")
+    @PostMapping("/min/{communityId}")
     ResponseEntity<List<BorrowRequest>> getBorrowRequestsInCommunityInAmountRange(
             @RequestParam(value = "minAmount", required = false) BigDecimal minAmount,
-            @RequestParam(value = "maxAmount", required = false) BigDecimal maxAmount,
+            @RequestParam(value = "maxAmount", required = true) BigDecimal maxAmount,
             @RequestParam(value = "lessThan", required = false, defaultValue = "false") boolean lessThan,
-            @RequestParam(value = "greaterThan", required = false, defaultValue = "true") boolean greaterThan
+            @RequestParam(value = "greaterThan", required = false, defaultValue = "true") boolean greaterThan,
+            @PathVariable String communityId
     ) throws ControllerException {
         try{
+            log.info("minAmount: {}", minAmount);
+            log.info("maxAmount: {}", maxAmount);
+            log.info("lessThan: {}", lessThan);
+            log.info("greaterThan: {}", greaterThan);
+            log.info("communityId: {}", communityId);
             List<BorrowRequest> borrowRequests = new ArrayList<>();
-
             if (minAmount != null && maxAmount != null) {
-                if (minAmount.compareTo(maxAmount) >= 0) {
+                if (minAmount.compareTo(maxAmount) > 0) {
                     log.error("minAmount is greater than maxAmount which is not permissible");
                     return ResponseEntity.badRequest().body(borrowRequests);
                 }
             }
-
             else if(minAmount==null && maxAmount==null){
                 log.error("Both minAmount and maxAmount are empty");
                 return ResponseEntity.badRequest().body(borrowRequests);
             }
-
             if(lessThan && greaterThan){
+                log.error("Inaccurate Constrains");
                 return ResponseEntity.badRequest().body(borrowRequests);
             }
             else if(lessThan){
-                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityLessThanAmount(maxAmount);
+                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityLessThanAmount(maxAmount, communityId);
             }
             else if(greaterThan){
-                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityGreaterThanAmount(minAmount);
+                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityGreaterThanAmount(minAmount, communityId);
             }
             else{
-                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityInRange(minAmount,maxAmount);
+                borrowRequests=borrowRequestService.getBorrowRequestsInCommunityInRange(minAmount,maxAmount, communityId);
             }
```

Comment:
Adding too many if else cases is always a bad practice, 
It increases the cyclomatic complexity and make your code less testable.